KBaseMetagenomes.spec

```diff
@@ -3,16 +3,20 @@
 
 module KBaseMetagenomes {
 
-    /*
-    Data type for binned contigs:
-
-    typedef struct {
-        ...
-
-    } BinnedContigs;
+    typedef string bin_id;
+    typedef string contig_id;
 
+    /*
+     Reference to an assembly object
+     @id ws KBaseGenomeAnnotations.Assembly
     */
+    typedef string contig_obj_ref;
 
+    typedef structure {
+        list<bin_id> bin_ids;
+        mapping<bin_id, list<contig_id>> bin2contig_id;
+        mapping<contig_id, contig_obj_ref> contig_id2contig_ref;
```

Comment:
Actually, thinking about this more, this is probably not the right way to go.  There will be potentially hundreds of thousands of contigs, so this will create hundreds of thousands of references potentally, which even if they are all the same (and checked as one), I think will slow down saving the object to the WS since they will all have to be rewritten and validated.  It will also increase the size of the data in the WS.